### PR TITLE
feat: add project skeleton

### PR DESCRIPTION
**Pull request by johan-lindqvist**
## Tasklist
- [x] Remove serverless template stuff and use CDK instead

### CODE REVIEW

File: .globalconfig [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+# Function parameters use camelCase
+# https://docs.microsoft.com/dotnet/standard/design-guidelines/naming-parameters
+dotnet_naming_symbols.parameters_group.applicable_kinds = parameter
+dotnet_naming_rule.parameters_rule.symbols              = parameters_group
+dotnet_naming_rule.parameters_rule.style                = camel_case_style
+dotnet_naming_rule.parameters_rule.severity             = warning
+
+# File header template
+file_header_template = -------------------------------------------------------------------\n Copyright (c) Axis Communications AB, SWEDEN. All rights reserved.\n-------------------------------------------------------------------
```
Comment (ahult) [inline .globalconfig:343]: Remove this 😂

File: infrastructure/Cdk/Cdk.csproj [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (ahult) [inline infrastructure/Cdk/Cdk.csproj:5]: 7.0?

File: infrastructure/Cdk/GlobalSuppressions.cs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Potential Code Quality Issues", "RECS0026:Possible unassigned object created by 'new'", Justification = "Constructs add themselves to the scope in which they are created")]
+[assembly: SuppressMessage("Naming", "CA1711:Identifiers should not have incorrect suffix", Justification = "<Pending>", Scope = "namespaceanddescendants", Target = "~N:Cdk")]
```
Comment (ahult) [inline infrastructure/Cdk/GlobalSuppressions.cs:4]: Add justification

File: infrastructure/Cdk/Program.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+namespace Cdk;
+
+public sealed class Program
+{
+	public static void Main(string[] args)
+	{
+		var app = new App();
+		_ = new RaidmateStack(app, "RaidmateStack", new StackProps
+		{
+			StackName = "RaidmateBackendStack"
```
Comment (ahult) [inline infrastructure/Cdk/Program.cs:12]: I think this should have a suffix/prefix for env? (prod, stage, devname, etc)

File: src/RaidmateBackend/Controllers/LoginController.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		var request = new HttpRequestMessage(HttpMethod.Post, "https://oauth.battle.net/token");
+		request.Headers.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.UTF8.GetBytes($"{clientId}:{clientSecret}")));
+		request.Headers.TryAddWithoutValidation("cache-control", "no-cache");
+		request.Headers.TryAddWithoutValidation("content-type", "application/x-www-form-urlencoded");
+
+		var parameters = new Dictionary<string, string>
+		{
+			{ "scope", "wow.profile" },
+			//{ "redirect_uri", "https://localhost:53846/login/redirect" },
+			{ "redirect_uri", "http://localhost:8080/blizzard-login" },
```
Comment (ahult) [inline src/RaidmateBackend/Controllers/LoginController.cs:59]: is this the redirect back to the website? should prob be a part of configuration, or actually, maybe sent in from the client?

File: infrastructure/Cdk/Cdk.csproj (hunk 2 above)
Comment (johan-lindqvist) [reply infrastructure/Cdk/Cdk.csproj:5]: Lambda doesn't have native support for .NET 7. They have custom runtimes but I've never used them. We could try out the custom runtimes later, but just wanted to get it up and running for now.

File: infrastructure/Cdk/Program.cs (hunk 4 above)
Comment (johan-lindqvist) [reply infrastructure/Cdk/Program.cs:12]: #8

File: src/RaidmateBackend/Controllers/LoginController.cs (hunk 5 above)
Comment (johan-lindqvist) [reply src/RaidmateBackend/Controllers/LoginController.cs:59]: #9

File: infrastructure/Cdk/Cdk.csproj (hunk 2 above)
Comment (johan-lindqvist) [reply infrastructure/Cdk/Cdk.csproj:5]: #10

**Pull request merged**